### PR TITLE
gh-86943: implement `pathlib.WindowsPath.is_mount()`

### PR DESCRIPTION
**Pull request by barneygale**
Have `pathlib.WindowsPath.is_mount()` call `ntpath.ismount()`. Previously it raised `NotImplementedError` unconditionally.

https://bugs.python.org/issue42777

* Issue: gh-86943

Automerge-Triggered-By: GH:brettcannon

### CODE REVIEW

File: Doc/library/pathlib.rst [hunk 1]
```diff
@@ -847,10 +847,15 @@ call fails (for example because the path doesn't exist).
    function checks whether *path*'s parent, :file:`path/..`, is on a different
    device than *path*, or whether :file:`path/..` and *path* point to the same
    i-node on the same device --- this should detect mount points for all Unix
-   and POSIX variants.  Not implemented on Windows.
+   and POSIX variants.  On Windows, a mount point is considered to be a drive
+   letter root (e.g. ``c:\``), a UNC share (e.g. ``\\server\share``), or a
+   volume mounted on a filesystem folder.
```
Comment (eryksun) [inline Doc/library/pathlib.rst]: Could you change the last case to make it more generic, such as  "a mounted filesystem directory"? I'd prefer to keep it open to a broader interpretation. The phrase "a volume mounted on" to me implies that the entire namespace of the volume is mounted on the directory. That can change, and I think it should change. In POSIX, `ismount()` makes no distinction between a volume mount and a bind mount.

**Comment by Ovsyanka83** [comment]
Will review soon

**Comment by Ovsyanka83** [comment]
Will review soon

**Comment by barneygale** [comment]
> Will review soon

Don't worry about it - I'm trying to focus my efforts on two more important PRs: #31691 and #91882. The branch has conflicts anyway.

**Comment by barneygale** [comment]
Not sure what's going on with this failure - looks like `ntpath.ismount('C:/\x00')` is true in `main`? It raises an exception in my Windows VM with Python 3.10.

**Comment by eryksun** [comment]
> Not sure what's going on with the test failure - looks like `ntpath.ismount('C:/\x00')` is true in `main`? It raises an exception in my Windows VM with Python 3.10.

The new implementation of `normpath()` strips the trailing null character. The old implementation keeps it.

**Comment by eryksun** [comment]
> The new implementation of `normpath()` strips the trailing null character. The old implementation keeps it.

Do you think this needs to be fixed, or is it an improvement?

**Comment by barneygale** [comment]
> Do you think this needs to be fixed, or is it an improvement?

I think it's unlikely to come up in real world usage. IMO it's one of the quirks noted in 99fcf150521:

> This commit attempts to preserve every previously tested quirk, but these may be changed in the future to better align platforms.

**Review by brettcannon — APPROVED** [review]

**Review by brettcannon — APPROVED** [review]

**Pull request merged**